### DISCUSSION TITLE
Commander send_setpoint(roll,pitch,yawrate,thrust)

### DISCUSSION THREAD

**Discussion by Andret997** [bitcraze/discussions#869] [Q&A]
hi i'm trying to send some setpoints through the commander with the method send_setpoint(roll,pitch,yawrate,thrust) with the future idea of making a controller on the position.
i have a problem with sending this setpoint to my crazyflie which fails to communicate instead with all the other methods in commander.py i have sent the correct setpoint.
I believe this is because send_setpoint uses pk.port=CRTPPort.COMMANDER whereas all other methods use CRTPPort.COMMANDER_GENERIC.
how can i fix it?
why is the pitch angle reversed in -pitch?
Thanks, Andrea

**Comment by krichardsson** [comment]
[`send_setpoint(roll,pitch,yawrate,thrust)`](https://github.com/bitcraze/crazyflie-lib-python/blob/294e9cb7f6b7d5c7afb0fb1a1147f2201f81779c/cflib/crazyflie/commander.py#L68-L88) is used from the python client [when flying with a gamepad](https://github.com/bitcraze/crazyflie-clients-python/blob/d387b70d3f96be8c72f749ec4de5722f582bc2c5/src/cfclient/ui/main.py#L196-L198), you can check that it works by flying manually.

There is also [an example](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/motors/ramp.py) you can try that uses the function.

> i have a problem with sending this setpoint to my crazyflie which fails to communicate

In what way does it fail to communicate? What happens? Can you explain a bit more what you want to do, what you expect to happen and what you observe?

**Comment by krichardsson** [comment]
> why is the pitch angle reversed in -pitch?

Historical reasons...

There are various quirks in how things are defined in the Crazyflie that dates back to when the project started. It is a pain to change but it might happen some day... also see https://github.com/bitcraze/crazyflie-firmware/issues/537
